### PR TITLE
fix(driving-license-duplicate): Application overhaul

### PR DESCRIPTION
**Pull request by kksteini**
## What

* Update application to use modern style of dataproviders
* Update current license provider to support v5 (legacy support via parameters provided so far)
* Update application to serve newer requirements
* Update application to submit to an endpoint

## Screenshots / Gifs

[Screencast from fös 23.jún 2023 13:57:28.webm](https://github.com/island-is/island.is/assets/77672665/7c31c6f6-fc99-4fb5-afb1-1741438e5aa5)

## Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Formatting passes locally with my changes
- [X] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/template-api-modules/src/lib/modules/templates/driving-license-duplicate/driving-license-duplicate.service.ts [hunk 1]
```diff
@@ -3,37 +3,41 @@ import { DrivingLicenseService } from '@island.is/api/domains/driving-license'
...
+    super(ApplicationTypes.DRIVING_LICENSE_DUPLICATE)
+  }
 
   async createCharge({
     application: { id, answers },
     auth,
   }: TemplateApiModuleActionProps) {
-    const SYSLUMADUR_NATIONAL_ID = '6509142520'
-
     // TODO: Change to AY116 once its available on dev until then use the regular drivingLicnese code
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/driving-license-duplicate/driving-license-duplicate.service.ts:30]:
```suggestion
```

File: libs/application/template-api-modules/src/lib/modules/templates/driving-license-duplicate/driving-license-duplicate.service.ts [hunk 2]
```diff
@@ -65,11 +69,22 @@ export class DrivingLicenseDuplicateService {
       }
     }
 
-    // TODO: SUBMIT functionality once the police update their api
-
+    // TODO: add reason to dublicate submission?
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/driving-license-duplicate/driving-license-duplicate.service.ts:72]:
```suggestion
    // TODO: add reason to duplicate submission?
```

File: libs/application/template-api-modules/src/lib/modules/templates/driving-license-duplicate/driving-license-duplicate.service.ts [hunk 3]
```diff
@@ -65,11 +69,22 @@ export class DrivingLicenseDuplicateService {
...
+    // Currently we are tracking "stolen" vs "lost" in the application
+    // Does this need to be tracked in the license system?
+    const orderId = await this.drivingLicenseService
+      .drivingLicenseDuplicateSubmission({
+        districtId: parseInt(answers.district.toString(), 10),
+        ssn: nationalId,
+      })
+      .catch((e) => {
+        // TODO: stop logging this here
+        console.log(JSON.stringify(e, null, 2))
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/driving-license-duplicate/driving-license-duplicate.service.ts:82]:
```suggestion
```

File: libs/application/templates/driving-license-duplicate/src/fields/PaymentCharge/index.tsx [hunk 4]
```diff
@@ -5,18 +5,58 @@ import { Box } from '@island.is/island-ui/core'
...
+  // Change price based on age, takes precedence over temporary license
+  // and therefore comes after it for overriding purposes
+  let age = info(application.applicant).age
+  if (allowFakeCondition(YES)(application.answers)) {
+    age = parseInt(
+      getValueViaPath<string>(application.answers, 'fakeData.age') ?? '25',
+      10,
+    )
+  }
+  if (age > 65) {
```
Comment (kksteini) [inline libs/application/templates/driving-license-duplicate/src/fields/PaymentCharge/index.tsx:51]:
```suggestion
  if (age >= 65) {
```

File: libs/application/templates/driving-license-duplicate/src/fields/PaymentCharge/index.tsx [hunk 5]
```diff
@@ -5,18 +5,58 @@ import { Box } from '@island.is/island-ui/core'
...
+import { YES } from '../../lib/constants'
+import { info } from 'kennitala'
+import { DriversLicense } from '@island.is/clients/driving-license'
 
 export const PaymentCharge: FC<FieldBaseProps> = ({ application }) => {
   const { formatMessage } = useLocale()
   const { setValue } = useFormContext()
-  const chargeCode = 'AY110'
+
+  let chargeCode = 'AY110'
```
Comment (kksteini) [inline libs/application/templates/driving-license-duplicate/src/fields/PaymentCharge/index.tsx:19]: Maybe comment what these chargeCodes are/represent

**Comment by kksteini** [comment]
Go over `allowFakeCondition` again.
Instead of `allowFakeCondition(NO)` þá kannski `!(allowFakeCondition(YES) || allowFakeCondition(IGNORE))`

Make sure all logic holds when fake condition is not set.

File: libs/application/templates/driving-license-duplicate/src/lib/messages.ts [hunk 6]
```diff
@@ -1,5 +1,29 @@
 import { defineMessages } from 'react-intl'
 export const applicationForMessages = {
+  A_FULL: defineMessages({
```
Comment (kksteini) [inline libs/application/templates/driving-license-duplicate/src/lib/messages.ts:3]: All of these applicationForMessages are redudnant now. Consider deleting them.

Comment (kksteini) [reply libs/application/templates/driving-license-duplicate/src/lib/messages.ts:3]: Deleted

File: libs/application/templates/driving-license-duplicate/src/fields/PaymentCharge/index.tsx (hunk 5 above)
Comment (kksteini) [reply libs/application/templates/driving-license-duplicate/src/fields/PaymentCharge/index.tsx:19]: Added

**Review by Thorrik58 — APPROVED** [review]
LGTM for codeowner files

**Review by alexdiljar — APPROVED** [review]
lgtm

**Pull request merged**